ts/Base/types.d.ts

```diff
@@ -0,0 +1,11 @@
+export type Diff<
```

Comment:
Just saw #918 shouldn't `types.d.ts` be `types.ts`?

Reply:
could do it, did `.d.ts` because it's just typing stuff, but I think that's not how it works